### PR TITLE
'Comprehension constructions'

### PR DESCRIPTION
**Pull request by ann-kolosova**
Create 'list_target ' from two lists by using comprehension constructions'

### CODE REVIEW

File: Lesson06_Task01.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+to get larger indexes scope ( or use if-else)
+dict as you remember has default value if key was not found d1.get(key, 0)
+l1 = [2, 4, 6, 8, 10]
+l2 = [1, 2, 3]
+"""
+
+l1 = [2, 4, 6, 8, 10]
+l2 = [1, 2, 3]
+
+([l1.append(0) for _ in range(len(l2) - len(l1))]
```
Comment (antohaUa) [inline Lesson06_Task01.py:27]: Key moment - we should NOT modify initial object this is best practice.
So better to create new object and leave initial objects as is.

Comment (antohaUa) [inline Lesson06_Task01.py:27]: You see such code looks too complicated better somehow to split it to more readable constructions or use another approach. 
Here is example from me ( just try to read it and understand what is going on there):
```
default = 0
d1 = {key: val for key, val in enumerate(l1)}
d2 = {key: val for key, val in enumerate(l2)}
l3 = [(d1.get(key, default), d2.get(key, default)) for key in set(d1.keys() | d2.keys())]
print(l3) 
```

**Review by antohaUa — APPROVED** [review]

File: Lesson06_Task01.py (hunk 1 above)
Comment (ann-kolosova) [reply Lesson06_Task01.py]: fixed

Comment (ann-kolosova) [reply Lesson06_Task01.py]: fixed

File: Lesson06_Task01.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+dict as you remember has default value if key was not found d1.get(key, 0)
+l1 = [2, 4, 6, 8, 10]
+l2 = [1, 2, 3]
+"""
+
+l1 = [2, 4, 6, 8, 10]
+l2 = [1, 2, 3]
+
+# copy l1, l2
+l3 = l1[:]
```
Comment (antohaUa) [inline Lesson06_Task01.py:28]: Better to do naming that will help you to understand that this is copy 
For example l1_copy = l1[:]

File: Lesson06_Task01.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+l4 = l2[:]
+
+# determine the lengths of lists
+len_l3 = len(l3)
+len_l4 = len(l4)
+
+# add 0s to list with shorter length
+if len_l3 < len_l4:
+    l3 += [0] * (len_l4 - len_l3)
```
Comment (antohaUa) [inline Lesson06_Task01.py:37]: 0 better to store as constant. For example tomorrow we want to fill default value as -1

**Review by antohaUa — APPROVED** [review]

File: Lesson06_Task01.py (hunk 2 above)
Comment (ann-kolosova) [reply Lesson06_Task01.py]: fixed

File: Lesson06_Task01.py (hunk 3 above)
Comment (ann-kolosova) [reply Lesson06_Task01.py]: fixed

**Pull request merged**